### PR TITLE
Move PL310 config from plat-stm to common code

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
Signed-off-by: Pascal Brand <pascal.brand@st.com>

### CODE REVIEW

File: core/arch/arm/kernel/tz_ssvce_pl310_a32.S [hunk 1]
```diff
@@ -24,21 +24,159 @@
...
+
+/*
+ * Memory Cache Level2 Configuration Function
+ *
+ * Use scratables registers R0-R3.
+ * No stack usage.
+ * LR store return address.
+ * Trap CPU in case of error.
+ */
+FUNC arm_cl2_config , :
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/tz_ssvce_pl310_a32.S]: This function is not generic

File: core/arch/arm/kernel/tz_ssvce_pl310_a32.S [hunk 2]
```diff
@@ -24,21 +24,159 @@
...
+	movt r1, #CPUID_A9_R3P0_H
+	cmp  r0, r1
+	beq  _config_l2cc_r3p0
+	b . /* TODO: unknown id: reset? log? */
+
+_config_l2cc_r3p0:
+	/*
+	 * TAG RAM Control Register
+	 *
+	 * bit[10:8]:1 - 2 cycle of write accesses latency
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/tz_ssvce_pl310_a32.S]: Isn't this configuration very platform specific?

File: core/arch/arm/kernel/tz_ssvce_pl310_a32.S [hunk 3]
```diff
@@ -24,21 +24,159 @@
...
+	 *
+	 * I/Dcache prefetch enabled (bit29:28=2b11)
+	 * NS can access interrupts (bit27=1)
+	 * NS can lockown cache lines (bit26=1)
+	 * Pseudo-random replacement policy (bit25=0)
+	 * Force write allocated (default)
+	 * Shared attribute internally ignored (bit22=1, bit13=0)
+	 * Parity disabled (bit21=0)
+	 * Event monitor disabled (bit20=0)
+	 * 128kB ways, 8-way associativity (bit19:17=3b100 bit16=0)
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/tz_ssvce_pl310_a32.S]: This looks like something very platform specific.

**Comment by pascal-brand-st-dev** [comment]
Ok. Then let's abandon this PR.

**Pull request closed without merging**